lib/datadog/statsd/forwarder.rb

```diff
@@ -55,7 +55,7 @@ def initialize(
           overflowing_stategy: buffer_overflowing_stategy,
         )
 
-        @sender = single_thread ? SingleThreadSender.new(buffer) : Sender.new(buffer)
+        @sender = single_thread ? SingleThreadSender.new(buffer) : Sender.new(buffer, logger: logger)
```

Comment:
Minor: I suggest keeping the interface the same between both implementations, including in the constructor here.

One other suggestion to make sure we always use both classes in the same way would be to do: 

```ruby
@sender = (single_thread ? SingleThreadSender : Sender).new(buffer, logger: logger)
```